presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragmenter.java

```diff
@@ -443,9 +443,6 @@ public PlanNode visitTableScan(TableScanNode node, RewriteContext<FragmentProper
         @Override
         public PlanNode visitTableWriter(TableWriterNode node, RewriteContext<FragmentProperties> context)
         {
-            if (node.getPartitioningScheme().isPresent()) {
```

Comment:
As i understand that was always extra? Right?

Reply:
From the message of the commit that introduced this method, it seems to be a sanity check assuming that we won't change the fragment distribution before table writer (https://github.com/prestodb/presto/commit/2323e95ef15ce4d97e441f5b6003e0fa6c931cb4), which is no longer true. This makes me wonder if we should just remove this `visitTableWriter` method entirely.

Reply:
> This makes me wonder if we should just remove this visitTableWriter method entirely.

Yeah, let's remote it altogether.